### DISCUSSION TITLE
Proposal: Allow null coalescing operator for pointers

### DISCUSSION THREAD

**Discussion by gafter** [dotnet/csharplang#323] [General]
_From @jacekbe on October 11, 2015 11:6_

I realize that pointers in C# are not the most widely used construct but they exist and it would be nice if null coalescing operator worked for pointer types as well so we could write:
`T *p1 = p2 ?? p3;`
instead of:
`T *p1 = p2 != null ? p2 : p3;`

_Copied from original issue: dotnet/roslyn#5869_

**Comment by gafter** [comment]
_From @aluanhaddad on October 11, 2015 22:35_

This makes sense. If `T *p1 = p2 != null ? p2 : p3;` works so should `T *p1 = p2 ?? p3;`

**Comment by lachbaer** [comment]
Would #328 be of use here? I have already implemented that into Roslyn as a *default-coalescing-operator*. Haven't tried though whether it also works for pointers.

Update: ~my solution currently does not work with pointers. I'm going to adopt it.~ working with pointer also now.

**Comment by jacekbe** [comment]
Default value for pointer types is null so your proposal would fit here.

**Comment by gafter** [comment]
I've queued this to be considered by the LDM at https://github.com/dotnet/csharplang/issues/418.
